src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -175,6 +175,10 @@ internal void Apply(Publishers publishers, UnicastRoutingTable unicastRoutingTab
             {
                 m.Configure((type, endpointAddress) =>
                 {
+                    if (!conventions.IsMessageType(type))
```

Comment:
Why is this check needed? Isn't that already covered by the APIs calling this method?